modules/kwin.nix

```diff
@@ -564,6 +571,175 @@ in
           };
         };
       };
+
+      krohnkite = {
+        enable = lib.mkOption {
+          type = with lib.types; nullOr bool;
+          default = null;
+          example = true;
+          description = "Whether to enable Krohnkite.";
+        };
+
+        settings = {
+          geometry = {
+            gaps = {
+              top = lib.mkOption {
+                type = with lib.types; nullOr (ints.between 0 999);
+                default = null;
+                example = 8;
+                description = "Top screen gap for krohnkite.";
+              };
+              left = lib.mkOption {
+                type = with lib.types; nullOr (ints.between 0 999);
+                default = null;
+                example = 8;
+                description = "Left screen gap for krohnkite.";
+              };
+              right = lib.mkOption {
+                type = with lib.types; nullOr (ints.between 0 999);
+                default = null;
+                example = 8;
+                description = "Right screen gap for krohnkite.";
+              };
+              bottom = lib.mkOption {
+                type = with lib.types; nullOr (ints.between 0 999);
+                default = null;
+                example = 8;
+                description = "Bottom screen gap for krohnkite.";
+              };
+              tiles = lib.mkOption {
+                type = with lib.types; nullOr (ints.between 0 999);
+                default = null;
+                example = 8;
+                description = "Gap between tiles for krohnkite.";
+              };
+            };
+
+            tileWidthLimit = {
+              enable = lib.mkOption {
+                type = with lib.types; nullOr bool;
+                default = null;
+                example = true;
+                description = "Whether to limit tile width for krohnkite.";
+              };
+              ratio = lib.mkOption {
+                type = with lib.types; nullOr (ints.between 1 100);
+                default = null;
+                example = 1.6;
+                description = "Tile width limiting ratio for krohnkite.";
+              };
+            };
+          };
+
+          layouts = {
```

Comment:
Perhaps this should also be an option that takes in a `oneOf` string value.

Reply:
could you provide an example or additional context

Reply:
LGTM.